xm.js

```diff
@@ -364,8 +365,8 @@ function MixChannelIntoBuf(ch, start, end, dataL, dataR) {
   var volE = ch.volE / 64.0;    // current volume envelope
   var panE = 4*(ch.panE - 32);  // current panning envelope
   var p = panE + ch.pan - 128;  // final pan
-  var volL = volE * (128 - p) * ch.vol / 8192.0;
-  var volR = volE * (128 + p) * ch.vol / 8192.0;
+  var volL = volE * (player.xm.global_volume - p) * ch.vol / 8192.0;
+  var volR = volE * (player.xm.global_volume + p) * ch.vol / 8192.0;
```

Comment:
I don't believe this is correct. If panning is 255 (hard right), then even if global volume is 0 it will still play on the right channel. The 128 is baked into panning and the 8192.0 in the denominator here.

I was going to implement it by setting the value of the gainNode created in the web audio graph. So just use `gainNode.gain.value = 0.1 * player.xm.global_volume / 128.0`.